src/stravalib/client.py

```diff
@@ -700,6 +771,12 @@ def create_activity(
         if distance is not None:
             params["distance"] = distance
 
+        params = self._validate_activity_type(
+            params, activity_type, sport_type, create=True
+        )
+        # TODO: throw warning if activity or sport_type not included
+        # in returned params
```

Comment:
Let `_validate_activity_type()` raise an error if neither are provided, as you indicated above:

> i set both of these "types" to optional. but we probably want to run a test on params that one or the other exists otherwise the user will get a bad call return

Reply:
Ah, now I see. I was indeed missing something 🙈. Both types are optional when updating, but at least one needs to be provided when creating. Your option 2 looks the best, as this check is specific to the create method.

Reply:
ok wonderful. i wasn't certain what you were thinking but we're on the same page now!

fixed locally. thanks!